public/app/plugins/panel/graph/graph.ts

```diff
@@ -925,6 +925,10 @@ class GraphElement {
       return formattedValueToString(formatter(val, axis.tickDecimals, axis.scaledDecimals));
     };
   }
+
+  canAddAnnotations() {
+    return (this.dashboard.meta.canEdit || this.dashboard.meta.canMakeEditable) && this.dashboard.editable;
```

Comment:
this.dashboard.editable should be here, then you cannot add annotations to dashboards that are not editable (which you should be able to do)

Reply:
Sorry I don't understand, but `this.dashboard.editable` is part of the logic?

Reply:
yes, think (this.dashboard.meta.canEdit || this.dashboard.meta.canMakeEditable)  should cover it

Reply:
It doesn't, unfortunately. It's very confusing with all these props. But when `editable = false` then an editor could still add annotations. See #29990.

Reply:
I'll do another test run to confirm

Reply:
I did another test run without `this.dashboard.editable` in the statement and it fails for non editable dashboards. For non editable dashboards someone with edit permissions would still be able to add annotations so `this.dashboard.editable` still needs to be in the statement.

Reply:
But that seems to be the current and correct behavior (that you can still add annotations to noneditable dashboards):
https://play.grafana.org/d/Z9NCtxBMz/test-editable?orgId=1

Reply:
Probably because #29990 hasn't been released yet.

Reply:
Hmm, now I'm a bit confused. Then this issue is not a correct then https://github.com/grafana/grafana/issues/28738

Reply:
yea, think the bug in https://github.com/grafana/grafana/issues/28738 was not that you cannot add an annotation but that you cannot then delete or edit it

Reply:
Ah I see, I'll try to fix this in this PR as well. Thank you for clarifing this.

Reply:
I've moved the `canAddAnnotations` function to the `DashboardModel` instead and added a check for this in annotation_tooltip as well.

Reply:
This means that 7.4.0-beta1 will have the intermittent behavior from #29990 until this PR gets merged. 
How do we do with back port labels and betas? Should I also add a back port label to this PR @torkelo?

Reply:
yes, that is needed